azkaban-common/src/main/java/azkaban/executor/ExecutorLoader.java

```diff
@@ -42,8 +42,12 @@ List<ExecutableFlow> fetchRecentlyFinishedFlows(Duration maxAge)
   Map<Integer, Pair<ExecutionReference, ExecutableFlow>> fetchActiveFlows(DispatchMethod dispatchMethod)
       throws ExecutorManagerException;
 
+  Pair<ExecutionReference, ExecutableFlow> fetchUnfinishedFlow(final int executionId)
+      throws ExecutorManagerException;
   Map<Integer, Pair<ExecutionReference, ExecutableFlow>> fetchUnfinishedFlows()
       throws ExecutorManagerException;
+  List<Integer> selectUnfinishedFlows(final int projectId, final String flowId) throws ExecutorManagerException;
+  List<Integer> selectUnfinishedFlows() throws ExecutorManagerException;
```

Comment:
We split fetchUnfinishedFlows into 4 different functions and let other classes to consume